src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2525,39 +2525,95 @@ end
 
 function _permutation_helper!(fun::Union{typeof(Base.permute!!), typeof(Base.invpermute!!)},
                               df::AbstractDataFrame, p::AbstractVector{<:Integer})
-    toskip = Set{Int}()
+    length(p) <= nrow(df) || throw(ArgumentError("Permutation is too long"))
```

Comment:
@nalimilan - what do you think we have two options:
1. leave what @LilithHafner proposes - then we need to update the docstring explaining that shorter permutations are allowed.
2. do `==` test and require permutation to have an exact length?

Reply:
Oh, I misread the tests and thought that we already supported small permutations previously and needed to maintain that. I don't see much benefit to supporting short vectors, especially if they have one-based indexing. Seems better to throw an error. Aside: In the vein of short vectors, it might be nice to specialize on sparse vectors, but this PR already does a good job of handling almost-identity transformations efficiently.

I vote 2.

Reply:
OK - so can you please change it to 2 (i.e. require `==`)

Reply:
I have pushed the change to approve the PR.